src/Helper/PaymentHelper.php

```diff
@@ -34,19 +53,126 @@ class PaymentHelper {
   public function __construct(
     private readonly RequestStack $requestStack,
     private readonly ClientInterface $httpClient,
+    private readonly EntityTypeManagerInterface $entityTypeManager,
+    private readonly LoggerChannelFactoryInterface $loggerFactory,
```

Comment:
Consider injection a specific logger (rather than the logger factory). See https://github.com/itk-dev/ai-screening/blob/develop/web/modules/custom/ai_screening/ai_screening.services.yml and https://github.com/itk-dev/ai-screening/blob/develop/web/modules/custom/ai_screening/src/Helper/AbstractHelper.php for an example.